watertap/ui/fsapi.py

```diff
@@ -438,41 +470,97 @@ def from_csv(self, file: Union[str, Path], flowsheet):
-                data["ui_units"] = pyo.units.dimensionless
-            else:
-                try:
-                    data["ui_units"] = eval(norm_units, {"units": pyo.units})
-                except Exception as err:
-                    raise ValueError(f"Bad units '{norm_units}': {err}")
+            # parse into object
+            data["obj"] = self._parse_object_text(data["obj"], flowsheet)
+            data["ui_units"] = self._parse_units_text(data["ui_units"])
             # process boolean values (starting with 'is_')
             for k in data:
                 if k.startswith("is_"):
-                    v = data[k].lower()
-                    if v == "true":
-                        data[k] = True
-                    elif v == "false":
-                        data[k] = False
-                    else:
-                        raise ValueError(
-                            f"Bad value '{data[k]}' "
-                            f"for boolean argument '{k}': "
-                            f"must be 'true' or 'false' "
-                            f"(case-insensitive)"
-                        )
+                    data[k] = self._parse_boolean_text(data[k], flowsheet)
             # add parsed export
+            # TODO: Mark output variables as deferred
             self.add(data=data)
             num += 1
 
         return num
 
+    @staticmethod
+    def _parse_object_text(text: str, flowsheet: Any) -> Any:
+        """Evaluate the object represented by 'text' in the context of the flowsheet.
+
+        Args:
+            text: Object starts with prefix 'fs' and then is an arbitrary combination
+                  of attributes and indices to indicate a variable/etc. within the flowsheet.
+            flowsheet: Flowsheet object substituted for 'fs.'
+
+        Returns:
+            Pyomo block/var/etc. within the flowsheet for 'text'
+
+        Raises:
+            ValueError, if the object named by 'text' could not be found
+        """
+        _log.debug(f"eval text={text}")
+        try:
+            obj = eval(text, {"fs": flowsheet})
+        except Exception as err:
+            raise ValueError(f"Cannot find object in flowsheet: {text}")
+
+        return obj
+
+    @staticmethod
+    def _parse_units_text(text: str) -> Any:
+        """Parse Pyomo units provided as text.
+
+        Whitespace is stripped before processing.
+
+        Three special values are allowed to mean 'dimensionless':
+        "" (empty string), "none", and "-".
+
+        Args:
+            text: Units, which begin with a prefix 'units.'
+
+        Returns:
+            Pyomo units object.
+
+        Raises:
+            ValueError, if bad units are provided
+        """
+        # evaluate the units
+        norm_units = text.strip()
+        if norm_units in ("", "none", "-"):
+            parsed_units = pyo.units.dimensionless
+        else:
+            try:
+                parsed_units = eval(norm_units, {"units": pyo.units})
```

Comment:
I think we could get rid of both the `eval()` and the need for a `units.` prefix is we used e.g. `pyo.units.get_units()`, which takes a string in input (the underlying Pint library is pretty good at parsing units from strings):

```suggestion
                parsed_units = pyo.units.get_units(norm_units)
```